import-service/src/functions/importFileParser/handler.ts

```diff
@@ -14,52 +18,60 @@ import { Handler, S3Event } from 'aws-lambda';
 import createError from 'http-errors';
 
 const importFileParser: Handler<S3Event> = async (event) => {
-  const eventObjectKey = event.Records[0].s3.object.key;
+  for await (const { s3 } of event.Records) {
+    const s3ObjectKey = s3.object.key;
+
+    const BUCKET = 'magic-files';
 
-  const BUCKET = 'magic-files';
+    const client = new S3Client({ region: 'eu-west-1' });
```

Comment:
You create a new connection to S3 inside a loop. It's better to move it on the top of the file and reuse it even between lambda function invocations.

Reply:
Yes, stupid enough. I planned to separate some lambdas from handlers, but this case is even worse) thank you

Reply:
fixed